autoload/flog.vim

```diff
@@ -1025,6 +1025,180 @@ endfunction
+" If it has been called before with a commit
+" present in the list, or has previously returned
+" an element of the current list, *that* element will be
+" returned. In addition, if the current commit is
+" not in the list of prior inputs and outputs,
+" that list is cleared. Hence there can only be one
+" matching element.
+" Use this as a replacement for just doing commit_list[0],
+" giving the current commit as an additional input. 
+" In the base case, it just returns
+" the first element. But the most recent inputs and outputs 
+" of this function are given priority before any others,
+" giving the function 'stability' when navigating up/down
+" a directed acyclic graph
+fu! flog#stable_select(commit_list, current_commit) abort
+  if index(g:flog_visited_commits, a:current_commit) == -1
+    let g:flog_visited_commits = [a:current_commit]
+  endif
+  let pick = flog#find_predicate(a:commit_list, {target_commit -> index(g:flog_visited_commits, target_commit) != -1})
+  if pick == v:null
+    let pick = a:commit_list[0]
+  endif
+  call add(g:flog_visited_commits, l:pick)
+  return pick
+endfunction
+
+fu! flog#jump_up_N_parents(amount) abort
+  let l:current_commit = flog#get_full_commit_hash()
+  if type(l:current_commit) != v:t_string
+    return v:null
+  endif
+  let c = 0
+  while c < a:amount
+    let l:parent_commit = system('git rev-list --parents -n 1 ' . l:current_commit)
+    let l:parents = split(l:parent_commit)[1:]
+    if len(l:parents) == 0
+      return
+    endif
+    let l:current_commit = flog#stable_select(l:parents, l:current_commit)
+    call flog#jump_to_commit(l:current_commit)
+    let c += 1
+  endwhile
+endfunction
+
+fu! flog#jump_to_parent() abort
+  call flog#jump_up_N_parents(v:count1)
+endfunction
+
+fu! flog#jump_down_N_children(amount) abort
+  let l:current_commit = flog#get_full_commit_hash()
+  if type(l:current_commit) != v:t_string
+    return v:null
+  endif
+  let c = 0
+  while c < a:amount
+    let l:child_commit = system("git log --format='%H %P' --all --reflog | grep -F \" " . l:current_commit . "\" | cut -f1 -d' '")
+    let l:children = split(l:child_commit)
+    if len(l:children) == 0
+      return
+    endif
+    let l:current_commit = flog#stable_select(l:children, l:current_commit)
+    call flog#jump_to_commit(l:current_commit)
+    let c += 1
+  endwhile
+endfunction
+
+fu! flog#jump_to_child() abort
+  call flog#jump_down_N_children(v:count1)
+endfunction
+
+fu! flog#offset_head_hash() abort
+  return fugitive#RevParse('HEAD@{' . g:flog_head_offset . '}')
+endfunction
+
+fu! flog#jump_to_offset_head(offset) abort
+  let l:current_commit = flog#get_full_commit_hash()
+  if type(l:current_commit) != v:t_string
+    return v:null
+  endif
+  let l:current_head_commit = flog#offset_head_hash()
+  if g:flog_head_offset == 0 || l:current_commit == l:current_head_commit
+    let l:reflog_size = str2nr(substitute(system('git reflog | wc -l'), '\v\C\n$', '', '')) - 1
```

Comment:
`wc -l` not system agnostic

Reply:
indeed, the reflog output should be parsed in vimscript

Reply:
done